app/Conference.php

```diff
@@ -82,12 +82,18 @@ public function onlineFriends()
 
     public function makeIntroduction($username)
     {
-        $friend = new Friend([
-            'username' => $this->normalizeFriendName($username),
-            'type' => $this->isUpcoming() ? 'online' : 'new',
-            'introduction' => true,
-            'met' => ! $this->isUpcoming(),
-        ]);
+        $friend = Friend::firstOrNew(['username' => $this->normalizeFriendName($username)]);
+
+        if ($friend->id) {
```

Comment:
Yeah, gross as sin. About to push a second pass. If we come up with a better way of handling this, I'll use it forever (or until we find a new better way).